pkg/util/mount/types/types.go

```diff
@@ -30,6 +30,8 @@ type Interface interface {
 	// to be consistent.
 	List() ([]MountPoint, error)
 	// IsMountPointMatch determines if the mountpoint matches the dir.
+	// On Linux systems, this method handles the case where the mountpoint has
```

Comment:
Do we want to specifically call out implementation specifics in the interface comment? Maybe something more generic like "In some environments, the directory could be renamed if the mount point is stale".

Or, from a completely different angle, does this have to be exposed as an interface method? Or can we completely hide it in the implementation?

Reply:
yeah, there is where I need help. :) One of the review comments was "why is this needed?" and another was "don't mention implementation details." But the reason it's needed is because the implementation is different on Linux. So we could just call out what's different in the Linux implementations (`mount_linux.go`, `nsenter`), but that leaves the question of what to say here...

But you are right... from a different angle, this doesn't appear needed in the interface. It's only called in one place: https://github.com/kubernetes/kubernetes/blob/d951d19ec4d8ec829e83d146200ea91b664b6fa6/pkg/util/mount/mount.go#L252

It's completely internal. We need different implementations for Windows/Linux, but that doesn't mean it has to be in the interface. I'll explore that.